### PR TITLE
CIS 4.1.3

### PR DESCRIPTION
**Pull request by romulets**
### Summary of your changes

Implement CIS 4.1.3.

The implementation follows the from AzureCLI approach:
image

The implemented approach is strict, were a encryption protection must exist and all returned encryption protection should comply to the rule. I think it's valid because it makes no sense (nor is possible to configure it) multiple transparent data encryptions. But, the API returns a slice of encryption protectors, so from a reading perspective it seems like it's possible to have multiple. In practice I believe we will always have only encryption protector returned.

Example of return from the API:

```json
{
  "value": [
    {
      "kind": "azurekeyvault",
      "properties": {
        "serverKeyName": "transparent-encryption_transparent-encryption-pass_235c66b5a8e1462fa021f9c4276af36f",
        "serverKeyType": "AzureKeyVault",
        "uri": "https://transparent-encryption.vault.azure.net/keys/transparent-encryption-pass/235c66b5a8e1462fa021f9c4276af36f"
      },
      "id": "/subscriptions/ef111ee2-6c89-4b09-92c6-5c2321f888df/resourceGroups/evgb_group/providers/Microsoft.Sql/servers/pass-412/encryptionProtector/current",
      "name": "current",
      "type": "Microsoft.Sql/servers/encryptionProtector"
    }
  ]
}
```

### Testing

[SQL Server pass-412](https://portal.azure.com/#@elastic365.onmicrosoft.com/resource/subscriptions/ef111ee2-6c89-4b09-92c6-5c2321f888df/resourceGroups/evgb_group/providers/Microsoft.Sql/servers/pass-412/overview) has been configured with proper transparent encryption and passes

### Related Issues
- Related: https://github.com/elastic/cloudbeat/issues/1417

### Checklist
- [x] I have added tests that prove my fix is effective or that my feature works

#### Introducing a new rule?

- [x] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [x] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

### CODE REVIEW

**Comment by romulets** [comment]
Beaware: there is a infosec rule about keyvault changes. If you change a key, you might get pinged by infosec. You just need to tell them it was on purpose :)

File: resources/providers/azurelib/inventory/sql_server.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	return assets, nil
+}
+
+func convertEncryptionProtector(ep *armsql.EncryptionProtector, resourceGroup string, subID string) AzureAsset {
+	return AzureAsset{
+		Id:       deref(ep.ID),
+		Name:     deref(ep.Name),
+		Location: deref(ep.Location),
+		Properties: map[string]any{
```
Comment (moukoublen) [inline resources/providers/azurelib/inventory/sql_server.go:61]: [Note]
Azure sdk models have custom marshaling containing names and populating optional values. 

Even though double conversion  `struct` -> `json` -> `map[any]` is not great performance-wise, it has the advantage of not having to hardcode the json name in cloudbeat code (and also takes care of omitting optional if this is something we need). 


https://github.com/Azure/azure-sdk-for-go/blob/b7a02cc26d3822ae665c896c758b6f94f5ca94eb/sdk/resourcemanager/sql/armsql/models_serde.go#L4869-L4878

**Review by moukoublen — APPROVED** [review]

File: resources/providers/azurelib/inventory/sql_server.go (hunk 1 above)
Comment (romulets) [reply resources/providers/azurelib/inventory/sql_server.go:61]: I agree. It's good to make a decision and keep it consistent

File: resources/providers/azurelib/inventory/sql_server_test.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+			} else {
+				require.NoError(t, err)
+			}
+
+			require.ElementsMatch(t, tc.expectedAssets, got)
+		})
+	}
+}
+
+func ref[T any](v T) *T {
```
[the remark this answers is not in the document]
Comment (romulets) [reply resources/providers/azurelib/inventory/sql_server_test.go:208]: Will do in the other PR!

**Pull request merged**